test/tomo/operator/xray_trafo_test.py

```diff
@@ -30,28 +30,261 @@
 # Internal
 import odl
 from odl.util.testutils import almost_equal
+from odl.tomo import ASTRA_CUDA_AVAILABLE
+from odl.tomo.util.testutils import skip_if_no_astra_cuda
 
 
-@pytest.mark.xfail  # Expected to fail since scaling of adjoint is wrong.
-@pytest.mark.skipif("not odl.tomo.ASTRA_CUDA_AVAILABLE")
-def test_xray_trafo():
-    """Test discrete X-ray transformt using ASTRA with CUDA."""
+# TODO: increase test coverage and improve tests
+
+# DiscreteLp volume / reconstruction space
+xx = 5
+nn = 5
+# xx = 5.5
+# nn = 11
+# xx = 5.25
+# nn = 21
+discr_vol_space_2d = odl.uniform_discr([-xx] * 2, [xx] * 2, [nn] * 2,
+                                       dtype='float32')
+discr_vol_space_3d = odl.uniform_discr([-xx] * 3, [xx] * 3, [nn] * 3,
+                                       dtype='float32')
+
+# Angle grid
+angle_intvl = odl.Interval(0, 2 * np.pi) - np.pi / 4
+agrid = odl.uniform_sampling(angle_intvl, 4)
+astride = float(agrid.stride)
+num_angle = agrid.ntotal
+
+# Detector grid
+# yy = 11
+# mm = 11
+yy = 10.5
+mm = 2 * 21
+dparams_2d = odl.Interval(-yy, yy)
+dgrid_2d = odl.uniform_sampling(dparams_2d, mm)
+dparams_3d = odl.Rectangle([-yy, -yy], [yy, yy])
+dgrid_3d = odl.uniform_sampling(dparams_3d, [mm] * 2)
+
+# Distances
+src_radius = 1000
+det_radius = 500
+mag = (src_radius + det_radius) / src_radius
+
+# Slice index to print
+z_vol = np.round(discr_vol_space_3d.shape[2] / 2)
+y_proj = np.floor(dgrid_3d.shape[1] / 2)
+
+# Precision for adjoint test
+precision = 4
+
+# Print section
```

Comment:
Yes. This was to figure out the scaling behaviour. Once the rescaling is correct, I want to change the print statements into tests.